### DISCUSSION TITLE
Limit minio "objects" per team with SSO

### DISCUSSION THREAD

**Discussion by simplydreamy0** [minio/minio#20447] [Q&A] [answered]
Hi !

We are looking to deploy minio as our S3 solution in my company and are looking to integrate it with our already deployed SSO solution (keycloak). 

We have a few questions remaining, notably on the permissions for users. We would like that teams of users have their own "space" inside the minio, that they would be able to access through the console or minio client, so that they'll only be able to see and access objects that concerns them directly.

From what we've seen [from the documentation](https://min.io/docs/minio/linux/administration/identity-access-management/oidc-access-management.html#minio-external-identity-management-openid-access-control) we can map the JWT's policy claim to a policy in minio so we would have keycloak set the right policy for the user. And we can use the OIDC variable "jwt:groups" to check if a user is part of a group.
We were thinking about using [objects tagging](https://min.io/docs/minio/linux/administration/identity-access-management/policy-based-access-control.html?ref=blog.min.io#tag-based-policy-conditions) to condition objects access for each group. Meaning each object would have a tag (i.e "team") with a value positionned to the team that the user pushed with and in the policy only allow the user to list/get/modify objects that have the same "team" tag. A problem with this method remains is: is there a way to automatically assign a tag to an object (bucket/file/access and secret keys, etc...) based on the OIDC group the user creates it with ?

My question is: Is this a common way to do it ? are we going in the wrong direction and if so which one would be the best for this use case ? We've seen that [minio tenants](https://min.io/docs/minio/kubernetes/upstream/operations/install-deploy-manage/deploy-minio-tenant.html) exists but it looks like they require us to deploy some k8s, manage it and we dont know how well having multiple tenants integrates with replication (we're deploying on 3 AZ) since each tenant seems to be an entire instance of minio.

Any help/feedback is greatly appreciated !
thanks in advance

**Comment by klauspost** [comment, marked as answer]
There is no such functionality in the server that is ready to use out of the box. You are welcome to reach out on `sales@min.io` describing your use case in more detail, and we can look at what options are available.
